### ISSUE TITLE
Bugs should only be modified if they are Fedora/EPEL-only

### ISSUE THREAD

**Issue by mprpic** [fedora-infra/bodhi#1043]
Bodhi occasionally tries to update bugs it should not, such as this one:

https://bugzilla.redhat.com/show_bug.cgi?id=1383657#c6

This is a top-level CVE bug and its state should not be modified by Bodhi since it tracks the issue in products other than Fedora and EPEL. Bodhi could simply check that the component is not 'vulnerability' and product 'Security Response' and skip making any state changes if it is.

**Comment by bowlofeggs** [comment]
I agree that this is important from a "politeness" perspective. Bodhi could make sure that BZs that are referenced in updates reference Fedora or EPEL bugs at creation time, and could re-ensure those states before modifying the bugs (in case they changed since Bodhi last checked). I think it would be fairly straightforward to make such a change and would be much friendlier on our part towards Red Hat's Bugzilla instance ☺

**Comment by pcahyna** [comment]
I noticed a Fedora bug : https://bugzilla.redhat.com/show_bug.cgi?id=1463547 in a package that I maintain for RHEL. I therefore cloned the Fedora bug to RHEL to track the issue: https://bugzilla.redhat.com/show_bug.cgi?id=1463608 . Bodhi then commented to the RHEL bug about updates to Fedora packages, and even changed its state MODIFIED → ON_QA. Why does it happen? Is it because the latter bug depends on the former? Or because the latter is a clone of the former? How to prevent it? Is it an instance of this issue?

**Comment by ktdreyer** [comment]
@pcahyna in [your dist-git change](https://src.fedoraproject.org/rpms/libtommath/c/baf704e7cfb469cf3eef392966489ef9fb0ad66f?branch=master) you added the following %changelog entry:

```
 * Sun Jun 25 2017 Simone Caronni <negativo17@gmail.com> - 1.0-6
 - Update URL (#1463608, #1463547).
```

The Bodhi client that `fedpkg update` uses will simply parse the latest %changelog entry and add all of those BZs to the Bodhi update.

I wonder if we should implement product filtering on the client side (when `fedpkg update` is constructing the Bodhi update template file) and on the server side as well.

**Comment by pcahyna** [comment]
@ktdreyer Thanks for the explanation! I did not thought about the %changelog (and it was not my dist-git change). I think product filtering is the right thing.

**Comment by mattiaverga** [comment]
@bowlofeggs I can work on this.
In fact, it seems the `Bugzilla.modified()` method already checks if `bug.product` is in `bz_products` list, am I wrong to suppose that we only need to add the same check to the other methods? (what does the bz_products list include? The default in the config file is empty...)

About the check at creation time, I don't think that would be necessary: from what I can see, the javascript that provides the bug list associated with a package gets data from apps.fedoraproject.org and only proposes Fedora and EPEL bugs.

**Comment by bowlofeggs** [comment]
@mattiaverga Ah interesting, I'd never noticed that Bodhi did avoid messing with bugs with modified, but did not do so for other states. I think you might be right - we should just make it so Bodhi doesn't alter bugs in other states too (like the check in modified).

One question I am unsure about is whether Bodhi should comment on these updates (without changing states), or whether it really should just do nothing at all. I think I lean towards it doing nothing at all, but I welcome feedback from others.

Fedora's production instance does [configure](https://infrastructure.fedoraproject.org/cgit/ansible.git/tree/roles/bodhi2/base/templates/production.ini.j2?id=39acdd1e6126b6a31865960a173fe27d81a95c3f#n464) the ```bz_products``` setting to be non-empty:

```
bz_products = Fedora,Fedora EPEL
```

Thus I think we simply need to make it so that the other methods also leave those BZs alone, and to add some tests to make sure those changes work.

Thanks for the offer to help!

**Comment by fche** [comment]
IMHO commenting alone is harmless & maybe helpful.

**Comment by bowlofeggs** [comment]
A fix for this issue is planned to be included in the upcoming 3.9.0 release: #2477.

**Comment by bowlofeggs** [comment]
The patch for this issue was deployed to Fedora's production Bodhi instance yesterday.
